src/cls/IPM/Repo/Oras/PublishService.cls

```diff
@@ -0,0 +1,168 @@
+Class %IPM.Repo.Oras.PublishService Extends (%IPM.Repo.Oras.PackageService, %IPM.Repo.IPublishService)
+{
+
+Method PublishModule(pModule As %IPM.Repo.Remote.ModuleInfo) As %Boolean
+{
+    Set repo = pModule.Name
+    Set tag = pModule.VersionString
+
+    #; Use a temp directory
+    Set tempFile = ""
+    Set tempDirectory = ##class(%File).NormalizeDirectory("/tmp/" _ repo _ "/" _ tag _ "/")
+
+    #; Empty temp directory first
+    If ##class(%File).DirectoryExists(tempDirectory) {
+        Set tSC = ##class(%IPM.Utils.File).RemoveDirectoryTree(tempDirectory)
+        $$$ThrowOnError(tSC)
+    }
+
+    #; Create temp directory
+    Set tCreated = ##class(%File).CreateDirectoryChain(tempDirectory,.tReturnValue)
+    If 'tCreated {
+		$$$ThrowStatus($$$ERROR($$$GeneralError,$$$FormatText("Error creating directory %1: %2",tempDirectory,tReturnValue)))
+    }
+
+    #; Create tgz file in temp directory
+    Set tFileBinStream = ##class(%Stream.FileBinary).%New()
+    Set tFileBinStream.Filename = ##class(%File).TempFilename("tgz", "/tmp/" _ repo _ "/" _ tag _ "/", .return)
+    Do tFileBinStream.CopyFromAndSave(pModule.Payload)
+
+    #; Convert module.xml into string
+    Set moduleXML = ""
+    While (pModule.Manifest.AtEnd = 0) {
+        Set moduleXML = moduleXML _ pModule.Manifest.ReadLine(,.status)
+        $$$ThrowOnError(status)
+    }
+
+	#; Set metadata
+	#; see https://github.com/opencontainers/image-spec/blob/main/annotations.md
+	Set metadata = {}
+	Do metadata.%Set("org.opencontainers.image.title", pModule.Name)
```

Comment:
For this metadata used in both publishing and retrieval, would be good to have an ObjectScript class to represent it. You can define properties with non-alphanumeric characters using double quotes.
That class can extend %JSON.Adaptor.

Reply:
Done.